### PR TITLE
feat: add Stripe related tables and move to using TxIDs

### PR DESCRIPTION
**Pull request by maximousblk**
### Stripe tables

- The stipe tables are prefixed with `Stripe`.
- I added all the columns that stripe's api provides.
- It's possible we won't need all the new columns and we can drop them as needed.

### Migrate to TxID
- This pr also removes the `url` column on `File` table and replaces it with `txid`.
- This will allow us to dynamically switch between and possibly distribute the load among multiple gateways as required.
- This also reduces redundant/useless data in the url like the gateway address and file path, which consequently reduces db size.
- Another advantage is that we can now easily check if the transactions are confirmed and show an error/warning if the files are requested before they are confirmed (ref: nestdotland/nest.land#319) 
- One drawback to this is that now this schema only works with arweave. Previous schema could've worked fine with http gateways for alternative protocols like ipfs, dat, etc. (could prolly be solved by adapters) which would be great for self-hosted instances tho that is not a concern for now.

### CODE REVIEW

**Review by oganexon — CHANGES_REQUESTED** [review]
LGTM !
Could you sync this branch with the main branch before merging ?
Looking at the “Files changed” tab, there are a lot more changes than you did

File: prisma/schema.prisma [hunk 1]
```diff
@@ -1,9 +1,6 @@
 generator client {
-  provider = "prisma-client-js"
-}
-
-generator nexusPrisma {
-  provider = "nexus-prisma"
```
Comment (oganexon) [inline prisma/schema.prisma:6]: Could you add back the `nexusPrisma` generator ?

Comment (maximousblk) [reply prisma/schema.prisma:6]: `nexus-prisma` doesn't exist on any repo than gql so it causes errors while generating client, etc.

Comment (oganexon) [reply prisma/schema.prisma:6]: ok this is a problem... If we remove that then the gql repo won't work anymore
I have to figure this out

**Review by oganexon — APPROVED** [review]
You can merge it for now, we'll figure out later the generator issue

**Pull request merged**